url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/manager/panel/KitManagerPanel.java

```diff
@@ -0,0 +1,358 @@
+package manager.panel;
+
+import java.awt.BorderLayout;
+import java.awt.CardLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.GridLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+
+import javax.swing.DefaultComboBoxModel;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTabbedPane;
+import javax.swing.JTextField;
+import javax.swing.border.EtchedBorder;
+
+import factory.KitConfig;
+import factory.PartType;
+
+/*
+* Authorship: Aaron Harris and Matt Zecchini
+*/
+
+public class KitManagerPanel extends JPanel implements ActionListener {
+	private manager.KitManager km;
+	
+	private ArrayList<KitConfig> kitConfigs;
+	private ArrayList<PartType> partTypes;
+	
+	private JComboBox[] cbPart;
+	private JComboBox cbKits;
+	private Object[] partModel;
+	private DefaultComboBoxModel kitModel;
+	private JTextField tfName;
+	private JPanel pnlButtons;
+	private JButton btnAddKit;
+	private JButton btnEditKit;
+	private JButton btnDeleteKit;
+	private JButton btnCreateKit;
+	private JButton btnClrFields;
+	private JButton btnSaveChg;
+	private JButton btnCnclChg;
+	
+	private final PartType noPart = new PartType("No Part");
+	private JPanel pnlParts;
+
+	/**
+	 * Create the panel.
+	 */
+	public KitManagerPanel(manager.KitManager k) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 61 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 88-123 into a separate method.